Buddies/Controls/FilterSearchBar.swift

```diff
@@ -47,87 +53,200 @@ class FilterSearchBar : UISearchBar, UISearchBarDelegate {
+        let label = UILabel(frame: CGRect(x: 0, y: 0, width: 0, height: 0))
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = text
+        return label
     }
     
-    func renderNewFilterMenu() -> UIView {
-        let height = CGFloat(200)
-        let container = UIView(frame: CGRect(x: 0, y: 0, width: frame.width, height: height))
-        container.backgroundColor = UIColor(white: 0.98, alpha: 0.85)
+    func makeButton(saying text: String, doing action: Selector) -> UIButton {
+        let bttn = UIButton(type: .system)
+        bttn.setTitle(text, for: .normal)
+        bttn.translatesAutoresizingMaskIntoConstraints = false
+        bttn.addTarget(self, action: action, for: .touchUpInside)
+        return bttn
+    }
+    
+    func renderNewFilterMenu() -> FilterMenuElements {
+        let container = UIView(frame: CGRect(x: 0, y: 0, width: 0, height: 0))
+        container.backgroundColor = .clear
         container.translatesAutoresizingMaskIntoConstraints = false
-        
-        // Rounded corners
         container.clipsToBounds = true
         container.layer.cornerRadius = 10
         
-        // render
-        superview?.addSubview(container)
+        let blurView = UIVisualEffectView(effect: UIBlurEffect(style: .light))
+        blurView.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Make sliders
+        let dateSlider = makeSlider(from: 1, to: 6)
+        dateSlider.delegate = DateRangeSliderDelegate.instance
+        dateSlider.minDistance = 1
+        dateSlider.selectedMinValue = CGFloat(lastSearchParams.dateMin)
+        dateSlider.selectedMaxValue = CGFloat(lastSearchParams.dateMax)
+        
+        let locationRangeSlider = makeSlider(from: 1, to: 200)
+        locationRangeSlider.selectedMaxValue = lastSearchParams.maxMilesAway
+        locationRangeSlider.disableRange = true
+        
+        // Make labels
+        let dateSliderLabel = makeLabel(saying: "When:")
+        let locationSliderLabel = makeLabel(saying: "Distance (miles):")
+        
+        // Make buttons
+        let innerFilterButton = makeFilterButton(
+            doing: #selector(self.saveFilterMenu),
+            rounding: [
+                .layerMaxXMaxYCorner,
+                .layerMaxXMinYCorner,
+                .layerMinXMaxYCorner,
+                .layerMinXMinYCorner,
+            ])
+        let cancelButton = makeButton(saying: "Cancel", doing: #selector(self.closeFilterMenu))
+        
+        
+        
+        // put things together
+        let containerChildren = [
+            dateSliderLabel,
+            dateSlider,
+            locationSliderLabel,
+            locationRangeSlider,
+            innerFilterButton,
+            cancelButton,
+        ]
+        container.addSubview(blurView)
+        containerChildren.forEach { container.addSubview($0) }
+        
+        let parent = superview!.superview! // Get the window in an ugly way
```

Comment:
I also didn't like having to do this - It's like "fair" for like one level

Reply:
Pretty much impossible to avoid at this point, though

Reply:
oh this was supposed to go in my review lol. oops.

Reply:
oh and apparently all my other comments went singularly as well when I posted this one. oh well